src/main/kotlin/com/team4099/robot2023/subsystems/elevator/ElevatorIONeo.kt

```diff
@@ -0,0 +1,134 @@
+      followerSparkMax,
+      ElevatorConstants.GEAR_RATIO,
+      ElevatorConstants.SPOOL_RADIUS * 2,
+      ElevatorConstants.VOLTAGE_COMPENSATION
+    )
+
+  private val leaderPIDController: SparkMaxPIDController = leaderSparkMax.pidController
+  private val followerPIDController: SparkMaxPIDController = followerSparkMax.pidController
+
+  init {
+    leaderSparkMax.restoreFactoryDefaults()
+    followerSparkMax.restoreFactoryDefaults()
+
+    leaderSparkMax.clearFaults()
+    followerSparkMax.clearFaults()
+
+    leaderSparkMax.enableVoltageCompensation(ElevatorConstants.VOLTAGE_COMPENSATION.inVolts)
+    followerSparkMax.enableVoltageCompensation(ElevatorConstants.VOLTAGE_COMPENSATION.inVolts)
+
+    leaderSparkMax.inverted = ElevatorConstants.LEFT_MOTOR_INVERTED
+    followerSparkMax.inverted = ElevatorConstants.RIGHT_MOTOR_INVERTED
+
+    leaderSparkMax.setSmartCurrentLimit(ElevatorConstants.PHASE_CURRENT_LIMIT.inAmperes.toInt())
+    followerSparkMax.setSmartCurrentLimit(ElevatorConstants.PHASE_CURRENT_LIMIT.inAmperes.toInt())
+
+    // TODO(figure out if we need this)
+    leaderSparkMax.openLoopRampRate = ElevatorConstants.RAMP_RATE
+    followerSparkMax.openLoopRampRate = ElevatorConstants.RAMP_RATE
+
+    followerSparkMax.follow(leaderSparkMax)
+  }
+
+  override fun updateInputs(inputs: ElevatorIO.ElevatorInputs) {
+    // TODO(is there any reason to log follower aswell)
+    inputs.elevatorPosition = leaderSensor.position
+
+    inputs.elevatorVelocity = leaderSensor.velocity
+
+    inputs.leaderAppliedOutput = leaderSparkMax.appliedOutput
+
+    inputs.leaderStatorCurrent = leaderSparkMax.outputCurrent.amps
+
+    // BatteryVoltage * SupplyCurrent = percentOutput * BatteryVoltage * StatorCurrent
+    // AppliedVoltage = percentOutput * BatteryVoltage
+    // SupplyCurrent = (percentOutput * BatteryVoltage / BatteryVoltage) * StatorCurrent =
+    // percentOutput * statorCurrent
+
+    inputs.leaderSupplyCurrent = inputs.leaderStatorCurrent * leaderSparkMax.appliedOutput
+
+    inputs.leaderTempCelcius = leaderSparkMax.motorTemperature.celsius
+
+    inputs.followerAppliedOutput = followerSparkMax.appliedOutput
+
+    inputs.followerStatorCurrent = followerSparkMax.outputCurrent.amps
+
+    inputs.followerSupplyCurrent = inputs.followerStatorCurrent * followerSparkMax.appliedOutput
+
+    inputs.followerTempCelcius = followerSparkMax.motorTemperature.celsius
+  }
+
+  override fun setOpenLoop(percentOutput: Double) {
+    leaderSparkMax.set(percentOutput)
+  }
+
+  override fun setPosition(position: Length, feedforward: ElectricalPotential) {
+
+    leaderPIDController.setFF(feedforward.inVolts)
+    followerPIDController.setFF(feedforward.inVolts)
+
+    leaderPIDController.setReference(position.inMeters, CANSparkMax.ControlType.kPosition)
+  }
+
+  override fun zeroEncoder() {
+    leaderSparkMax.encoder.position = 0.0
+    followerSparkMax.encoder.position = 0.0
+  }
+
+  override fun configPID(
```

Comment:
this won't work. for us, it's easier to understand kP values as output/error where output is volts and error is meters but for the actual sparkmax controller, this number is in native units. take a look at SwerveIOFalcon to see how to turn this into native sensor units (hint you need to use the sensors you created above)